go/host/host.go

```diff
@@ -876,40 +882,47 @@ func (h *host) checkBlockForSecretResponse(block *types.Block) bool {
 // (2) There are gaps in the historical chain of batches. To avoid an inconsistent state (i.e. one where we have stored
 // a batch without its parent), we request the sequencer to resend the batches we've just received, plus any missing
 // historical batches, then discard the received batches. We will store all of these at once when we receive them
-func (h *host) handleBatches(encodedBatches *common.EncodedBatches) error {
-	var batches []*common.ExtBatch
-	err := rlp.DecodeBytes(*encodedBatches, &batches)
+func (h *host) handleBatches(encodedBatchMsg *common.EncodedBatchMsg) error {
+	var batchMsg *hostcommon.BatchMsg
+	err := rlp.DecodeBytes(*encodedBatchMsg, &batchMsg)
 	if err != nil {
 		return fmt.Errorf("could not decode batches using RLP. Cause: %w", err)
 	}
-	if len(batches) == 0 {
-		return nil
-	}
 
-	// We store the batches and submit them to the enclave.
-	for _, batch := range batches {
-		// TODO - #718 - Think carefully about the risk of inconsistency between the enclave and the host in terms of
-		//  batches stored.
-		if err = h.enclaveClient.SubmitBatch(batch); err != nil {
-			return fmt.Errorf("could not submit batch. Cause: %w", err)
-		}
+	for _, batch := range batchMsg.Batches {
+		// TODO - #718 - Consider moving to a model where the enclave manages the entire state, to avoid inconsistency.
 
-		err = h.batchManager.StoreBatch(batch)
+		// If we do not have the block the rollup is tied to, we skip processing the batches for now. We'll catch them
+		// up later, once we've received the L1 block.
+		_, err = h.db.GetBlockHeader(batch.Header.L1Proof)
 		if err != nil {
-			if !errors.Is(err, batchmanager.ErrBatchesMissing) {
-				return fmt.Errorf("could not store batches. Cause: %w", err)
+			if errors.Is(err, errutil.ErrNotFound) {
+				return nil
 			}
+			return fmt.Errorf("could not retrieve block. Cause: %w", err)
+		}
 
-			// We have encountered missing batches. We abort the storage operation and request the missing batches.
-			batchRequest, err := h.batchManager.CreateBatchRequest(h.config.P2PPublicAddress)
-			if err != nil {
-				return fmt.Errorf("could not create batch request. Cause: %w", err)
-			}
+		isParentStored, batchRequest, err := h.batchManager.IsParentStored(batch)
+		if err != nil {
+			return fmt.Errorf("could not determine whether batch parent was missing. Cause: %w", err)
+		}
+
+		// We have encountered a missing parent batch. We abort the storage operation and request the missing batches.
+		// We only do this if the batches did not themselves arrive as part of catch-up, to avoid excessive P2P pressure.
+		if !isParentStored && !batchMsg.IsCatchUp {
```

Comment:
Don't you need to also handle the case where `!isParentStored` but `isCatchup`? As it stands you send it to the enclave to fail I think.

Reply:
Fixed.